src/extension.js

```diff
@@ -10,7 +10,7 @@ exports.activate = function(context) {
     const document = editor.document;
 
     if (editor && document.languageId === "ruby" || document.languageId === "haml" || document.languageId === "erb") {
-      const pattern = /:?(["|']?(\w[^\w\s]?)+["|']?)+\s*?=>\s*/g
+      const pattern = /[\d|:((\w[^\w\s]?))]+\s*?=>\s*/g
```

Comment:
I think there are a few problems with this. When I paste it into a JavaScript console, I get a syntax error:

```
/[\d|:((\w[^\w\s]?))]+\s*?=>\s*/g
VM270:1 Uncaught SyntaxError: Invalid regular expression: /[\d|:((\w[^\w\s]?))]+\s*?=>\s*/: Unmatched ')'
    at <anonymous>:1:1
```

Also, the Ruby code in your example [here](https://github.com/Kosai106/vscode-ruby-syntax-replacer/issues/4#issuecomment-571185448) is syntactically invalid, since you can't use numbers as keys in Ruby hashes with the colon syntax:

```
[1] pry(main)> { 1: :lol }
SyntaxError: unexpected ':', expecting =>
{ 1: :lol }
```

Reply:
I'll admit I've not used Ruby for almost two years now, but the syntax I'm using in my examples all worked fine when I was testing, which is why I'm confused by all of this.

I'll gladly accept contributions if you have a better solution in mind.